osu.Game/Overlays/BeatmapSet/Header.cs

```diff
@@ -208,6 +210,18 @@ private void load(OsuColour colours)
                     favouriteButton.FadeOut(transition_duration);
                 }
 
+                if (setInfo.NewValue?.OnlineInfo.Availability?.DownloadDisabled ?? false)
```

Comment:
why not use the same pattern as all the other components and move this logic inside the `BeatmapNotAvailable` component itself?